libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -461,6 +479,28 @@ void CodeTransform::operator()(FunctionDefinition const& _function)
 		}
 	}
 
+	if (!subTransform.m_delayedReturnVariables.empty())
+	{
+		// Can only happen for functions with straight control flow (in particular no jump to the exit label)
+		// that never reads from or writes to return variables.
+
+		// We should have allocated slots either for all or for none of the return variables.
+		yulAssert(subTransform.m_delayedReturnVariables.size() == _function.returnVariables.size(), "");
+
+		// Already pop all arguments to make the stack shuffling below easier.
+		while (m_assembly.stackHeight() > (m_evm15 ? 0 : 1))
+			m_assembly.appendInstruction(evmasm::Instruction::POP);
+
+		for (auto const& _returnVariable: _function.returnVariables)
+		{
+			auto* var = std::get_if<Scope::Variable>(&varScope->identifiers.at(_returnVariable.name));
+			yulAssert(var, "Return variable not in scope.");
+			m_context->variableStackHeights[var] = static_cast<size_t>(m_assembly.stackHeight());
+			// Set unassigned return variables to zero.
+			m_assembly.appendConstant(u256(0));
```

Comment:
We could also delay that until the shuffling - but I guess this situation also does not happen often, does it?

Reply:
Yeah, it's mainly a safeguard for weird cases - I guess this mainly happens for dummy functions used in tests... usually I'd expect a function to do *something* that triggers return value initialization sooner or later.